source/tests/test.ts

```diff
@@ -582,15 +582,15 @@ test('is.plainObject', t => {
 });
 
 test('is.iterable', t => {
-	t.true(m.iterable(''));
-	t.true(m.iterable([]));
-	t.true(m.iterable(new Map()));
-	t.false(m.iterable(null));
-	t.false(m.iterable(undefined));
-	t.false(m.iterable(0));
-	t.false(m.iterable(NaN));
-	t.false(m.iterable(Infinity));
-	t.false(m.iterable({}));
+	t.true(m.iterable('' as any));
+	t.true(m.iterable([] as any));
+	t.true(m.iterable(new Map() as any));
+	t.false(m.iterable(null as any));
+	t.false(m.iterable(undefined as any));
+	t.false(m.iterable(0 as any));
+	t.false(m.iterable(NaN as any));
+	t.false(m.iterable(Infinity as any));
+	t.false(m.iterable({} as any));
```

Comment:
As explained above, this is the reason why I'm not sure if we should strongly type the `value` arguments. My gut feeling says `I just want to throw values to the library, and it should either return true or false`. Always having to typecast it to the specific type either assumes you already know what it is, or otherwise you need type it as `any`.

Another example would be a function which either returns a `Promise` or an `Observable`.

```ts
const value = someFunction();

// `value` is of type `Promise` | `Observable`, but have to cast it explicitely to `Promise` or any.
if (is.promise(value as Promise)) {
   // ...
}
```